droidlet/dialog/map_to_dialogue_object.py

```diff
@@ -35,12 +36,21 @@ def __init__(
     def get_dialogue_object(
         self, speaker: str, chat: str, parse: Dict, chat_status: str, chat_memid: str
     ):
-        """Returns DialogueObject for a given chat and logical form"""
+        """Returns DialogueObject (or ingredients for a DialogueTask) 
+        for a given chat and logical form"""
+
         # 1. If we are waiting on a response from the user (e.g.: an answer
         # to a clarification question asked), return None.
-        if (len(self.dialogue_manager.dialogue_stack) > 0) and (
+        dialogue_stack_busy = (len(self.dialogue_manager.dialogue_stack) > 0) and (
             self.dialogue_manager.dialogue_stack[-1].awaiting_response
-        ):
+        )
+        _, active_task_mems = self.dialogue_manager.memory.basic_search(
+            "SELECT MEMORY FROM Task WHERE prio > -1"
+        )
+        dialogue_task_busy = any(
+            [getattr(m, "awaiting_response", False) for m in active_task_mems]
+        )
+        if dialogue_stack_busy or dialogue_task_busy:
```

Comment:
`if dialogue_stack_busy or dialogue_task_busy:` doesn't seem right 
Did we mean `if dialogue_stack_busy or active_task_mems:` ?

Reply:
I think its right as is: it returns None, right away, if there is a dialogue Task awaiting a response or if there is a something on the stack awaiting a response.   that first thing is going away....

if there is an active dialogue task but it isn't awaiting a response, we continue